### PR TITLE
Upgrade borsh unstable dependency to v1.0 and make it stable

### PR DESCRIPTION
**Pull request by bgeron**
I checked manually that even with borsh =v1.0.0 the tests pass.

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -441,7 +441,7 @@ pub enum Variant {
 )]
 #[cfg_attr(
     all(uuid_unstable, feature = "borsh"),
```
Comment (KodrAus) [inline src/lib.rs:443]:
```suggestion
    feature = "borsh",
```

The `uuid_unstable` gate should no longer be necessary now that `borsh` is stable.

**Review by KodrAus — COMMENTED** [review]
Thanks @bgeron! It looks like `borsh` has been stable for a little while now, but still a huge props to the authors for pulling it all together 👏

**Comment by bgeron** [comment]
> Thanks @bgeron! It looks like `borsh` has been stable for a little while now, but still a huge props to the authors for pulling it all together 👏

Very much agree! :clap: I'm not involved in borsh but I'm thinking to start using it.

- I changed more to make feature `borsh` stable, per your suggestion.

- I noticed that it wasn't possible to compile `uuid` with only the `borsh` feature because it didn't yet imply `dep:borsh-derive`, so I added that implication. Now a dummy program that depends on both uuid and borsh works. I'm not so experienced with Cargo feature implication but it seems to work.

File: Cargo.toml [hunk 2]
```diff
@@ -71,6 +71,8 @@ fast-rng = ["rng", "rand"]
 sha1 = ["sha1_smol"]
 md5 = ["md-5"]
 
+borsh = ["dep:borsh", "dep:borsh-derive"]
```
Comment (KodrAus) [inline Cargo.toml:74]: The fact that this works on our MSRV now is great :+1: There's so much I can clean up elsewhere with our dependencies.

**Review by KodrAus — APPROVED** [review]
Thanks @bgeron! This looks good to me.

**Pull request merged**